framework/src/application.ts

```diff
@@ -229,6 +233,21 @@ export class Application {
 		this._blockchainDB = this._getDBInstance(this.config, 'blockchain.db');
 		this._nodeDB = this._getDBInstance(this.config, 'node.db');
 
+		// Acquire mutex for loading process
+		this._runMutexReleaseHandler = await this._mutex.acquire();
```

Comment:
i think no need to assign it to the instance variable here.
also, it's probably safer to run runExecutive all below code so that if something failed it will release immediately and run the cleanup.
Also, in that way, no need to think about the timeout. (ie: depending on the genesis block size, the node init might vary on time